src/db/migrate/20200405151412_create_saved_functions.rb

```diff
@@ -0,0 +1,10 @@
+class CreateSavedFunctions < ActiveRecord::Migration[6.0]
+  def change
+    create_table :saved_functions do |t|
+      t.references :user, null: false, foreign_key: true
+      t.references :function, null: false, foreign_key: true
```

Comment:
[R] we'll need a unique compound index on `user_id` and `function_id` to prevent duplicate saves on the DB level

Reply:
Do you suggest a name for this column?

Reply:
@nemoload I don't think it will be a new column, it will a compound index on both two coloumns to apply uniqueness validation on Database Level not only application level in ActiveRecord Validation

Reply:
aha like so:
```
t.index [:function, :user], unique: true
```